Kodi 18.9 & 19.1 crashes on Android TV 8 (2x Philips TVs)

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Kodi 18.9 & 19.1 crashes "randomly" after start after a few minutes.


## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
It should not crash.


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
After adding new media with emby kodi crashes out to the ATV home screen.
Quickmic (dev of emby) and i have tracked it down to the "SyncProgramsJobService.java" file.
Kodi Debug log is useless, because the android sync is not visible in there.
In the logcat, the following exception is visiable:

Kodi 19: https://paste.kodi.tv/kowutakuja.kodi
```
07-19 21:39:11.743  1259  1259 E hw-IPCThreadState: binder thread pool (1 threads) starved for 125 ms
07-19 21:39:11.753  2469  3724 E JavaBinder: *** Uncaught remote exception!  (Exceptions are not yet supported across processes.)
07-19 21:39:11.753  2469  3724 E JavaBinder: android.database.SQLException: Failed to insert row into content://android.media.tv/preview_program
07-19 21:39:11.753  2469  3724 E JavaBinder: 	at com.android.providers.tv.TvProvider.insert(TvProvider.java:1389)
07-19 21:39:11.753  2469  3724 E JavaBinder: 	at android.content.ContentProvider$Transport.insert(ContentProvider.java:268)
07-19 21:39:11.753  2469  3724 E JavaBinder: 	at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:152)
07-19 21:39:11.753  2469  3724 E JavaBinder: 	at android.os.Binder.execTransact(Binder.java:674)
--------- beginning of crash
07-19 21:39:11.784  5717  6886 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #5
07-19 21:39:11.784  5717  6886 E AndroidRuntime: Process: org.xbmc.kodi, PID: 5717
07-19 21:39:11.784  5717  6886 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at android.os.AsyncTask$3.done(AsyncTask.java:353)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.getLastPathSegment()' on a null object reference
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at android.content.ContentUris.parseId(ContentUris.java:85)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.createPrograms(SyncProgramsJobService.java:212)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.syncPrograms(SyncProgramsJobService.java:182)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.doInBackground(SyncProgramsJobService.java:139)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.doInBackground(SyncProgramsJobService.java:111)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at android.os.AsyncTask$2.call(AsyncTask.java:333)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
07-19 21:39:11.784  5717  6886 E AndroidRuntime: 	... 4 more
07-19 21:39:11.823  5717  6044 I Kodi    : 2021-07-19 21:39:11.823 T:6044     INFO <general>: INFO: EMBY.core.tvshows.TVShows: ADD episode [68/411/7250/7901] 2447673: Das Feuer der Finsternis
07-19 21:39:11.873  1259  1259 E hw-IPCThreadState: binder thread pool (1 threads) starved for 130 ms
07-19 21:39:11.893  1683  2073 D ActivityManager: debugdump : copycrashInfo org.xbmc.kodi
07-19 21:39:11.959  5717  6044 I Kodi    : 2021-07-19 21:39:11.959 T:6044     INFO <general>: INFO: EMBY.core.tvshows.TVShows: ADD episode [68/411/7251/7902] 2447674: Die Leere der Finsternis
07-19 21:39:11.999  1259  1259 E hw-IPCThreadState: binder thread pool (1 threads) starved for 126 ms
07-19 21:39:12.000  1683  1683 D HDMI    : [P]:AllocatedAddress=[5]
07-19 21:39:12.016  1278  1343 I debugdump: event->name javacrash#5717#org.xbmc.kodi & eventName 
07-19 21:39:12.016  1278  1343 I debugdump: Received file create notification under /data/javacrash folder. File name : javacrash#5717#org.xbmc.kodi
07-19 21:39:12.045  1683  1702 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
07-19 21:39:12.046  1683  1702 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.chimera.GmsIntentOperationService$PersistentTrustedReceiver
07-19 21:39:12.117  1278  1341 V debugdump: server: msg recevied : javacrash#5717#org.xbmc.kodi 28
07-19 21:39:12.124  1683  2073 W ActivityManager:   Force finishing activity org.xbmc.kodi/.Main
07-19 21:39:12.131  1278  1341 V debugdump: parseMsg: pid = 5717, type = 2, maintype =java , subtype = crash, processname = org.xbmc.kodi effect = visible
07-19 21:39:12.139  5717  6044 I Kodi    : 2021-07-19 21:39:12.139 T:6044     INFO <general>: INFO: EMBY.core.tvshows.TVShows: ADD episode [68/411/7252/7903] 2447675: Die Finsternis erblüht
07-19 21:39:12.170  5717  5913 V Kodi    : virtual void CXBMCApp::onPause(): 
07-19 21:39:12.170  5717  5913 V Kodi    : static bool CXBMCApp::EnableWakeLock(bool): false
07-19 21:39:12.191  5717  5717 D TvUtil  : Scheduled channel creation.
07-19 21:39:12.203  1278  1341 I debugdump: dumpId: dump_00000603_00000001_00001938.635722_5717
07-19 21:39:12.219  1278  1341 I debugdump: Initialized zip file
07-19 21:39:12.220  1278  1341 I debugdump: Copied from /data/javacrash/javacrash#5717#org.xbmc.kodi(size:1438) to javacrash.txt
07-19 21:39:12.244  5717  5913 V Kodi    : virtual void CXBMCApp::onLostFocus(): 
07-19 21:39:12.257  5717  6886 I Process : Sending signal. PID: 5717 SIG: 9
07-19 21:39:12.306  1278  1341 I debugdump: Copied Kernel buffer to dump folder
07-19 21:39:12.387  3946  6899 E ActivityThread: Failed to find provider info for tvlauncher.mic
07-19 21:39:12.388  2068  2068 I chatty  : uid=1000(system) org.droidtv.xtv expire 3 lines
07-19 21:39:12.430  1264  1720 E SurfaceFlinger: Failed to find layer (SurfaceView - org.xbmc.kodi/org.xbmc.kodi.Main#0) in layer parent (no-parent).
07-19 21:39:12.431  1683  1769 W InputDispatcher: channel '9e9c2be org.xbmc.kodi/org.xbmc.kodi.Main (server)' ~ Consumer closed input channel or an error occurred.  events=0x9
07-19 21:39:12.431  1683  1769 E InputDispatcher: channel '9e9c2be org.xbmc.kodi/org.xbmc.kodi.Main (server)' ~ Channel is unrecoverably broken and will be disposed!
07-19 21:39:12.431  1264  1720 E SurfaceFlinger: Failed to find layer (Background for - SurfaceView - org.xbmc.kodi/org.xbmc.kodi.Main#0) in layer parent (no-parent).
07-19 21:39:12.501  1683  5395 I WindowManager: WIN DEATH: Window{9e9c2be u0 org.xbmc.kodi/org.xbmc.kodi.Main}
07-19 21:39:12.502  1683  5395 W InputDispatcher: Attempted to unregister already unregistered input channel '9e9c2be org.xbmc.kodi/org.xbmc.kodi.Main (server)'
07-19 21:39:12.528  1683  1770 I InputReader: Reconfiguring input devices.  changes=0x00000004
07-19 21:39:12.530  1683  3967 D MediaSessionService: Media button session is changed to null
07-19 21:39:12.556  1683  1792 D NsdService: Client disconnected
07-19 21:39:12.559  1683  2101 I ActivityManager: Process org.xbmc.kodi (pid 5717) has died: vis  TOP 
```

And Kodi 18.9: https://paste.kodi.tv/hibofuvova
```
--------- beginning of crash
07-19 22:13:46.729  4182  5914 E AndroidRuntime: FATAL EXCEPTION: AsyncTask #4
07-19 22:13:46.729  4182  5914 E AndroidRuntime: Process: org.xbmc.kodi, PID: 4182
07-19 22:13:46.729  4182  5914 E AndroidRuntime: java.lang.RuntimeException: An error occurred while executing doInBackground()
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at android.os.AsyncTask$3.done(AsyncTask.java:353)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:271)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:245)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:764)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.getLastPathSegment()' on a null object reference
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at android.content.ContentUris.parseId(ContentUris.java:85)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.createPrograms(SyncProgramsJobService.java:211)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.syncPrograms(SyncProgramsJobService.java:181)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.doInBackground(SyncProgramsJobService.java:138)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at org.xbmc.kodi.channels.SyncProgramsJobService$SyncProgramsTask.doInBackground(SyncProgramsJobService.java:110)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at android.os.AsyncTask$2.call(AsyncTask.java:333)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
07-19 22:13:46.729  4182  5914 E AndroidRuntime: 	... 4 more
07-19 22:13:46.880  3695  5949 I DynamiteModule: Considering local module com.google.android.gms.cronet_dynamite:0 and remote module com.google.android.gms.cronet_dynamite:1411
07-19 22:13:46.880  3695  5949 I DynamiteModule: Selected remote version of com.google.android.gms.cronet_dynamite, version >= 1411
07-19 22:13:46.886  1712  3398 D ActivityManager: debugdump : copycrashInfo org.xbmc.kodi
07-19 22:13:46.894  3621  3842 W zygote  : Skipping duplicate class check due to unrecognized classloader
07-19 22:13:46.981  1265  1265 D audio_hal: [out_standby:486]stream = 0xb064bdc0
07-19 22:13:46.981  1265  1265 I audio_hal: [do_output_standby:429]output_type = 0 drain = 1 
07-19 22:13:46.985  1290  1355 I debugdump: event->name javacrash#4182#org.xbmc.kodi & eventName 
07-19 22:13:46.985  1290  1355 I debugdump: Received file create notification under /data/javacrash folder. File name : javacrash#4182#org.xbmc.kodi
07-19 22:13:46.988  1286  1286 D MTK_KL  : <4>[  159.747072] 
```

In the source file "https://github.com/xbmc/xbmc/blob/Matrix/tools/android/packaging/xbmc/src/channels/SyncProgramsJobService.java.in" on line 212 (kodi matrix) you can see the crashing line.
Sadly the next line is a log entry :-( so we dont know why it's crashing.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->

wrap a try catch around line 212 to 215:

`
        long programId = ContentUris.parseId(programUri);
        Log.d(TAG, "Inserted new program: " + programId);
        media.setProgramId(programId);
        mediasAdded.add(media);
`

**Can anybody build an matrix android test version with more logging around line 212?**
I dont have the skills to build it myself


### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. Start kodi
2. wait
3. kodi crashes


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
Kodi Debug log is useless, because the android sync is not visible in there.


### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->


## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->


### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ X] Android
 - [ ] iOS
 - [ ] tvOS
 - [ ] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Android TV 8 running on an Philips 65PUS8102 and 55PUS9104
 - Kodi version: 18.9 & 19.1


<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/tools/android/packaging/xbmc/src/channels/SyncProgramsJobService.java.in
+++ b/tools/android/packaging/xbmc/src/channels/SyncProgramsJobService.java.in
@@ -209,10 +209,14 @@ public class SyncProgramsJobService extends JobService
                         .insert(
                                 TvContractCompat.PreviewPrograms.CONTENT_URI,
                                 previewProgram.toContentValues());
-        long programId = ContentUris.parseId(programUri);
-        Log.d(TAG, "Inserted new program: " + programId);
-        media.setProgramId(programId);
-        mediasAdded.add(media);
+
+        if (programUri != null)
+        {
+          long programId = ContentUris.parseId(programUri);
+          Log.d(TAG, "Inserted new program: " + programId);
+          media.setProgramId(programId);
+          mediasAdded.add(media);
+        }
       }
 
       return mediasAdded;

